src/metabase/query_processor/middleware/limit.clj

```diff
@@ -24,7 +24,9 @@
   [query]
   (or (qp.constraints/max-results-bare-rows)
       (mbql.u/query->max-rows-limit query)
-      qp.i/absolute-max-results))
+      (if (some? (get-in query [:info :card-id]))
```

Comment:
Using `:info` is a no-go, see 
https://github.com/metabase/metabase/blob/be88a1b7d8af5f842a382ad5d2bab2836313b7f2/shared/src/metabase/mbql/schema.cljc#L1537

But what alternative is there? There's no other for the pre-processor to tell whether the query is a saved question or not.